mirgecom/viscous.py

```diff
@@ -0,0 +1,399 @@
+r""":mod:`mirgecom.viscous` provides helper functions for viscous flow.
+
+Viscous Flux Calculation
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. autofunction:: viscous_flux
+.. autofunction:: viscous_stress_tensor
+.. autofunction:: diffusive_flux
+.. autofunction:: conductive_heat_flux
+.. autofunction:: diffusive_heat_flux
+.. autofunction:: viscous_facial_flux
+
+Viscous Time Step Computation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. autofunction:: get_viscous_timestep
+.. autofunction:: get_viscous_cfl
+.. autofunction:: get_local_max_species_diffusivity
+"""
+
+__copyright__ = """
+Copyright (C) 2021 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import numpy as np
+from grudge.trace_pair import TracePair
+from meshmode.dof_array import thaw, DOFArray
+
+from mirgecom.flux import divergence_flux_central
+from mirgecom.fluid import (
+    velocity_gradient,
+    species_mass_fraction_gradient,
+    make_conserved
+)
+from mirgecom.eos import MixtureEOS
+
+
+def viscous_stress_tensor(discr, eos, cv, grad_cv):
+    r"""Compute the viscous stress tensor.
+
+    The viscous stress tensor $\tau$ is defined by:
+
+    .. math::
+
+        \mathbf{\tau} = \mu\left(\nabla{\mathbf{v}}
+        +\left(\nabla{\mathbf{v}}\right)^T\right) + (\mu_B - \frac{2\mu}{3})
+        \left(\nabla\cdot\mathbf{v}\right)
+
+    Parameters
+    ----------
+    discr: :class:`grudge.eager.EagerDGDiscretization`
```

Comment:
This should be `dcoll` and `grudge.DiscretizationCollection`. (Here and below)

Reply:
(#509) I added this to a housecleaning list because we need to update the interfaces, too, so that the documentation matches the interface.  But, when I tried to update the interfaces straightforwardly - it did not work.  There are some supporting changes needed, out-of-scope for this change set.